alexmenuet-alexmenuet-taulia/week2/day2/challenge/card_counting_alm.rb

```diff
@@ -0,0 +1,189 @@
+                @cards_dealt.push("- #{@deck[index]} -")
+# Here we only want the index 0 of the value of the card to be able to count                
+                array = @deck[index].split("-")
+# Keep track of the cards dealt to count them                
+                @cards_to_count.push(array[0])
+# Print this round of card dealt
+                print "- #{@deck[index]} -" 
+                index += 1
+# Wait 2 second to give time to the playre to count                
+                sleep 2 
+            end
+# Ask player to give his count            
+            request_count
+        end
+    end
+
+
+  ##################
+ ## Player Count ## 
+##################
+    def request_count
+        puts " "
+        puts "What's you count?"
+# Get player's count
+        user_count = gets.chomp
+        puts " "
+# Get my count        
+        @my_count = count(@cards_to_count)
+# Check if player is correct or not        
+        if @my_count.to_i == user_count.to_i
+            puts "my count is #{@my_count}, you are correct!"
+        else
+            puts "my count is #{@my_count}, what happened to you?!? ;)"
+        end
+        puts " "
+    end
+
+
+  ##############
+ ## My Count ## 
+##############   
+    def count(cards_to_count)
+        cards_to_count.each do |rank| 
+            case rank
+# -1 for higher cards            
+            when "10", "J", "Q", "K", "A"
+                @count -= 1
+                #p "minus #{@count} - #{rank}"
+            when "8", "9"
+                #Do nothing for those 2 guys
+# +1 for the rest of the cards
+            else
+                @count += 1
+                #p "plus #{@count} - #{rank}"
+            end
+        end
+        return @count
+   end
+end
+
+
+
+  ##################
+ ## Main program ## 
+##################
+# Just for fun request how many decks the player want to play with
+puts "How many decks do you wanna play with? (number between 1-10)"
+n = gets.chomp
+# Generate card deck
+deck = Card_game.new(n)
+#deck.show_deck
+out = false
+# While player want to play, stay in the game
+while out == false do
+# Make sure the player is ready to play    
+    puts "Ready to start counting? (Y/N)"
+    a = gets.chomp
+# If not, then end program    
+    if a.upcase == "N"
+        puts "Ok, bye, bye!"
+        out = true
+# If yes, then start the game        
+    elsif a.upcase == "Y"
+# Clear the terminal and set up the rules    
+        system "clear"
+        puts "Here are the rules!"
+        puts "  - Count starts off at #{n} - (#{n} * number_of_decks)."
+        puts "  - Cards 2 through 7 increase the count by 1."
+        puts "  - Count does not change if the card is an 8 or a 9."
+        puts "  - If the card is a 10, J, Q, K, or A, decrease the count by 1."
+        puts " "
+        print "Let's start counting..."
+        fin = Time.now + 3 
+        rest = 3
```

Comment:
This is an impressive program, Alex! With the exception of iterators, it is Ruby convention to use descriptive variable names. Others viewing your code should be able to infer the responsibility of each variable without you having to explain.